Curve-finance-suite/New-Fee/src/agent.ts

```diff
@@ -0,0 +1,63 @@
+/*
+The following is a transaction on the Ethereum mainnet that should create a finding
+
+https://etherscan.io/tx/0x0ccbffa23dff7d7c47dcdaed14c3d98aab5e4c63d531d8e243365745cbb1484e
+
+Run `npm run tx <TxId>` to test with a particular TX
+*/
+
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+} from "forta-agent";
+
+import {
+  provideEventCheckerHandler,
+  FindingGenerator,
+} from "forta-agent-tools";
+
+// The signature of the NewFee event
+export const NEW_FEE_EVENT_SIG = "NewFee(uint256,uint256)";
+
+// The address of the Pool Owner (PoolProxy.vy)
+const POOL_PROXY_ADDRESS = "0xeCb456EA5365865EbAb8a2661B0c503410e9B347";
```

Comment:
I checked this address and I couldn't find the event `NewFee` in the code. I think you'll need to use all the pool addresses from here `https://curve.readthedocs.io/ref-addresses.html#deployment-addresses`, or use the `addressesProvider` contract from Curve (fetching all the pools in this way could take some time), or change the approach to detect when the DAO calls `set_new_fee`

Reply:
The agent has been changed to detect a function call to the `Proxy Pool`s `apply_new_fee` function instead of listening for `NewFee` events.